### ISSUE TITLE
Invalid environment variables, rkt volume with dash or underscore in task name

### ISSUE THREAD

**Issue by blalor** [hashicorp/nomad#2358]
* Nomad v0.5.4
* CentOS Linux release 7.3.1611 (Core),  4.9.10-1.el7.elrepo.x86_64
* rkt Version: 1.24.0

```hcl
job "example" {
    group "example" {
        task "$TASK0" {
            driver = "rkt"

            config {
                image = "example.com/aci0:LATEST"

                port_map {
                    api = "api"
                }
            }
        }
        
        task "othertask" {
            driver = "rkt"
            
            config {
                image = "example.com/aci1:LATEST"
            }
    }
}
```

1. If `$TASK0` is set to `log-ingester`, `rkt run` will fail for the second (`othertask`) task because Nomad attempts to set the environment variable `NOMAD_ADDR_log-ingester_api`, and the `-` is not a valid environment variable name, per the [ACI spec](https://github.com/appc/spec/blob/master/spec/aci.md) ([IEEE Std 1003.1-2001](http://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap08.html)).

2. If `$TASK0` is set to `log_ingester` (trying to work around the above problem), `rkt run` will fail for the first (`log_ingester`/`$TASK0`) task because Nomad creates invalid volume names for the local, alloc, and secret directories:

        Error: invalid argument "$ALLOC_ID-log_ingester-alloc,kind=host,source=…" for --volume=$ALLOC_ID-log_ingester-alloc,kind=host,source=…:
            invalid value in --volume flag "$ALLOC_ID-log_ingester-alloc,kind=host,source=…":
                ACName must contain only lower case alphanumeric characters plus "-"

[ACName](https://github.com/appc/spec/blob/master/spec/types.md#ac-name-type) must be `^[a-z0-9]+([-][a-z0-9]+)*$`

The workaround for both of these is to make the task name match `^[a-z0-9]+$`.  The 2nd issue is obviously rkt-specific, but unless the Docker driver does something different about the environment variables, they're of limited use with bash, for example.

**Comment by blalor** [comment]
related to #2406 #2405

**Comment by schmichael** [comment]
1. is fixed by #2406 (there's an attached binary you can test with if you have time); working on 2. now.

My plan is to just escape underscores with dashes in rkt volume names since we just need a task unique name.

A safer approach would be to hash+base32 task names in volumes as the volume name just needs to be unique, but I'd really like to make the names human readable if possible.

**Comment by schmichael** [comment]
#2416 should enforce env var name compliance with the standard.
